src/main/java/org/opensearch/index/codec/customcodecs/CustomCodecService.java

```diff
@@ -52,6 +53,8 @@ public CustomCodecService(MapperService mapperService, IndexSettings indexSettin
         super(mapperService, indexSettings, logger);
         int compressionLevel = indexSettings.getValue(INDEX_CODEC_COMPRESSION_LEVEL_SETTING);
         final MapBuilder<String, Codec> codecs = MapBuilder.<String, Codec>newMapBuilder();
+        // Delegate to OpenSearch "default" codec
+        Supplier<Codec> defaultCodecSupplier = () -> super.defaultCodec();
```

Comment:
I believe you don't need `defaultCodecSupplier` anymore, just use `this::defaultCodec`